cub/test/test_device_adjacent_difference.cu

```diff
@@ -1,701 +0,0 @@
-
-struct DetectWrongDifference
-{
-  bool *flag;
-
-  __host__ __device__ DetectWrongDifference operator++() const
-  {
-    return *this;
-  }
-  __host__ __device__ DetectWrongDifference operator*() const
-  {
-    return *this;
-  }
-  template <typename Difference>
-  __host__ __device__ DetectWrongDifference operator+(Difference) const
-  {
-    return *this;
-  }
-  template <typename Index>
-  __host__ __device__ DetectWrongDifference operator[](Index) const
-  {
-    return *this;
-  }
-
-  __device__ void operator=(long long difference) const
-  {
-    if (difference != 1)
-    {
-      *flag = false;
-    }
-  }
-};
-
-void TestAdjacentDifferenceWithBigIndexesHelper(int magnitude)
-{
-  const std::size_t elements = 1ll << magnitude;
-
-  thrust::device_vector<bool> all_differences_correct(1, true);
-
-  thrust::counting_iterator<long long> in(1);
-
-  DetectWrongDifference out = {
-    thrust::raw_pointer_cast(all_differences_correct.data())
-  };
-
-  AdjacentDifferenceCopy<READ_LEFT>(in, out, cub::Difference{}, elements);
-  AssertEquals(all_differences_correct.front(), true);
-}
-
-
-void TestAdjacentDifferenceWithBigIndexes()
-{
-  TestAdjacentDifferenceWithBigIndexesHelper(30);
-  TestAdjacentDifferenceWithBigIndexesHelper(31);
-  TestAdjacentDifferenceWithBigIndexesHelper(32);
-  TestAdjacentDifferenceWithBigIndexesHelper(33);
-}
-
-struct InvocationsCounter
-{
-  int *m_d_counts{};
-
-  explicit InvocationsCounter(int *d_counts) : m_d_counts(d_counts) {}
-
-  __device__ int operator()(int l, int /* r */) const 
-  {
-    atomicAdd(m_d_counts + l, 1);
-    return l;
-  }
-};
-
-void TestAdjacentDifferenceOpInvocationsNum(int num_items)
-{
-  auto in = thrust::make_counting_iterator(0);
-  auto out = thrust::make_discard_iterator();
-
-  thrust::device_vector<int> num_of_invocations(num_items, 0);
-  InvocationsCounter op{thrust::raw_pointer_cast(num_of_invocations.data())};
-
-  AdjacentDifferenceCopy<READ_LEFT>(in, out, op, num_items);
-  AssertEquals(
-    num_items - 1,
-    thrust::count(num_of_invocations.begin() + 1, num_of_invocations.end(), 1));
-  AssertEquals(0, num_of_invocations[0]);
-
-  thrust::fill_n(num_of_invocations.begin(), num_items, 0);
-  AdjacentDifferenceCopy<READ_RIGHT>(in, out, op, num_items);
-  AssertEquals(
-    num_items - 1,
-    thrust::count(num_of_invocations.begin(), num_of_invocations.end() - 1, 1));
-  AssertEquals(0, num_of_invocations[num_items - 1]);
-}
-
-void TestAdjacentDifferenceOpInvocationsNum()
-{
-  for (int num_items = 1; num_items < 4096; num_items *= 2)
-  {
-    TestAdjacentDifferenceOpInvocationsNum(num_items);
-  }
-}
-
-int main(int argc, char** argv)
-{
-  CommandLineArgs args(argc, argv);
-
-  // Initialize device
-  CubDebugExit(args.DeviceInit());
-
-  TestSize(0);
-  for (std::size_t power_of_two = 2; power_of_two < 20; power_of_two += 2)
-  {
-    TestSize(1ull << power_of_two);
-  }
-  TestAdjacentDifferenceWithBigIndexes();
-  TestAdjacentDifferenceOpInvocationsNum();
```

Comment:
important: is there an alternative to the `TestAdjacentDifferenceOpInvocationsNum` test?